TradeItIosTicketSDK2/TradeItBrokerLogoService.swift

```diff
@@ -6,20 +6,54 @@ enum TradeItBrokerLogoSize: String {
 }
 
 class TradeItBrokerLogoService {
-    static func setLogo(forBroker broker: TradeItBroker, onImageView imageView: UIImageView, withSize size: TradeItBrokerLogoSize) {
-        guard let logoMetadata = broker.logos as? [TradeItBrokerLogo],
+    private var cache: [String: [TradeItBrokerLogoSize: UIImage]] = [:]
+
+    func loadLogo(
+        forBroker broker: TradeItBroker,
+        withSize size: TradeItBrokerLogoSize,
+        onSuccess: @escaping (UIImage) -> Void,
+        onFailure: @escaping () -> Void
+    ) {
+        guard let brokerName = broker.shortName,
+            let logoMetadata = broker.logos as? [TradeItBrokerLogo],
             let logoData = logoMetadata.first(where: { $0.name == size.rawValue }),
             let logoUrlString = logoData.url,
             let logoUrl = URL(string: logoUrlString) else {
-                return print("TradeIt ERROR: No broker logo provided for \(broker.shortName ?? "")")
+                return print("TradeIt Logo: No broker logo provided for \(broker.shortName ?? "")")
+        }
+
+        if let cachedImage = getCachedLogo(brokerName: brokerName, size: size) {
```

Comment:
Just to understand, the cache is because potentially we are fetching the same logo on the Welcome screen and the Select broker screen ?